cirq-google/cirq_google/optimizers/two_qubit_gates/__init__.py

```diff
@@ -1,5 +1,15 @@
 # pylint: disable=wrong-or-nonexistent-copyright-notice
+from cirq import _compat
+
 from cirq_google.optimizers.two_qubit_gates.gate_compilation import (
     gate_product_tabulation,
     GateTabulation,
 )
+
+_compat.deprecated_submodule(
```

Comment:
Once you've added this deprecation, the old module should be directly moved to the new location. Right now, you seem to have copied the code in `cirq-google/cirq_google/optimizers/two_qubit_gates/math_utils.py` to `cirq-core/cirq/transformers/heuristic_decompositions/gate_tabulation_math_utils.py` and original module remains as it is. 

You should do a 

```
> git mv cirq-google/cirq_google/optimizers/two_qubit_gates/math_utils.py cirq-core/cirq/transformers/heuristic_decompositions/gate_tabulation_math_utils.py
> git mv cirq-google/cirq_google/optimizers/two_qubit_gates/math_utils_test.py cirq-core/cirq/transformers/heuristic_decompositions/gate_tabulation_math_utils_test.py
```


Once this is done, you can simply add a test to the `gate_tabulation_math_utils_test.py` to verify that references to old module are working as expected. 

See https://github.com/quantumlib/Cirq/pull/4799 for a reference.

Reply:
I see, thanks for the link. My bad, I didn't realise that was the way deprecation worked for modules.

Regarding test - considering that the code was moved rom cirq-google to cirq-core, is it even valid to have a test in cirq-core that refers to cirq-google functionality?

Reply:
Ah, that's right. My bad. We shouldn't add the test to cirq-core. We can probably just the deprecation test to `gate_compilation_test.py` which tests for deprecation warning of `math_utils.py`. Then both `math_utils.py` and `math_utils_test.py` can be moved using `git mv` preserving the commit history.

Reply:
Thanks for the quick reply, test added.

As far as I could see (e.g. check 'Files changed' tab), git already knows that these files were moved. AFAIU git got confused due to the original files still being around.